hbase-server/src/main/java/org/apache/hadoop/hbase/io/FSDataInputStreamWrapper.java

```diff
@@ -270,22 +271,18 @@ public void unbuffer() {
       if (this.instanceOfCanUnbuffer == null) {
         // To ensure we compute whether the stream is instance of CanUnbuffer only once.
         this.instanceOfCanUnbuffer = false;
-        Class<?>[] streamInterfaces = streamClass.getInterfaces();
-        for (Class c : streamInterfaces) {
-          if (c.getCanonicalName().toString().equals("org.apache.hadoop.fs.CanUnbuffer")) {
-            try {
-              this.unbuffer = streamClass.getDeclaredMethod("unbuffer");
-            } catch (NoSuchMethodException | SecurityException e) {
-              if (isLogTraceEnabled) {
-                LOG.trace("Failed to find 'unbuffer' method in class " + streamClass
-                    + " . So there may be a TCP socket connection "
-                    + "left open in CLOSE_WAIT state.", e);
-              }
-              return;
+        if (CanUnbuffer.class.isAssignableFrom(streamClass)) {
```

Comment:
@jojochuang Thanks for review! I tried your suggestion with the UT and it can pass. In my opinion, this changes should be OK, because the class implements the CanUnbuffer should implements the unbuffer() method. And have the 
`if (CanUnbuffer.class.isAssignableFrom(streamClass))` which make sure the
 `((CanUnbuffer) wrappedStream).unbuffer(); ` can be invoked. Now, just tried the UT, I want to try the changes on a real env, after testing on real env worked well, then will update the commit.

Reply:
@jojochuang I submit a new commit use `((CanUnbuffer) wrappedStream).unbuffer(); ` and tested in my 6 nodes cluster the hbase read worked well. So I think your suggestion makes sense. Good suggestion! Thanks!

Reply:
I think why we use isAssignableFrom is for hadoop-2.7-? Why not just use` if (wrappedStream instanceof CanUnbuffer)`?

Reply:
@Apache9 Good question! Because here we cast the sub class to parent class/interface, generally if want to check the up cast, we use `isAssignableFrom`. Also here the logic we want to judge the Class inheritance relationship, NOT the instance. Through use `(wrappedStream instanceof CanUnbuffer)` should get the same results, but I prefer `isAssignableFrom`.

Reply:
I think here we just want to confirm that it is a CanUnbuffer?

Reply:
NO, I think the logic is to confirm whether the stream implements the CanUnbuffer. :)

Reply:
@Apache9 Any more comments? If no more, could you help merge the commit? Thanks!

Reply:
Since CanUnbuffer is an interface, doesn't this just come down to syntax?

Reply:
@joshelser Which syntax you suggest,  the `instanceof` or `isAssignableFrom`?  The `instanceof` can be used to judge both class and interface instance, I think generally we used to judge the class instance, so here I prefer `isAssignableFrom`. Anymore question please feel free point out. Thanks!

Reply:
My point is not one of preference in syntax, just that there isn't any correctness problem between the alternatives. Because CanUnbuffer is an interface, all possible invocations of `instanceof` will be the same as `isAssignableFrom`. Please correct me if I'm wrong.

I am fine with what you have already, but it sounds like Duo prefers just `instanceof`.

Reply:
@joshelser Thanks for response! I agree with you. @jojochuang Seems we can NOT reach an agreement, could you help? Thanks!